site/docs/metadata.md

```diff
@@ -0,0 +1,134 @@
+# Metadata Tables
```

Comment:
What's the purpose of this documentation? Much of this is already covered in the [Spark queries page](https://iceberg.apache.org/spark-queries/#inspecting-tables), which this links to for docs on how to query the metadata tables.

Who is the intended audience for these docs?

Reply:
Because this is not a Spark only feature, the intention is to make it a top level documentation, and have proper tables for the metadata table schema instead of showing Java code.

Reply:
There is a pretty good argument against adding this, though. Documenting what tables exist without an engine is confusing to anyone looking for how to view metadata tables. Just saying that these tables exist and what their schemas are doesn't help a user coming to the docs.

If this were part of a document on the API that explained how to use the metadata tables and was targeted at engine developers, I think it would be more valuable. But I think placing it under tables will just cause confusion.

Reply:
Yes I agree that is a confusion point. So do you recommend having one section for each engine around system tables? My current thought is to have this page introducing the Iceberg schema, linking to related sections in each engine page for examples of using system tables.

One of my intention to add it was because this is a very important feature not exist in other similar products, and it provides huge benefits for users to build data management capabilities around such information. Tables like `manifests` and `files` also support optimizations like predicate pushdown and file pruning, which essentially solves the big metadata issue. So I feel it's a pity to hide such information too deeply.

Reply:
Yeah, I think for now we should include this for each engine. They do expose the tables differently (like Trino's `$` syntax) so I don't see a lot of value in splitting docs into common and engine-specific. That just makes it harder to find what you're looking for.